File: src/ServerPulse.Backend/AnalyzerApi/Domain/Dtos/ServerStatisticsResponse.cs
﻿namespace AnalyzerApi.Domain.Dtos
{
    public class ServerStatisticsResponse
    {
        public bool IsAlive { get; set; }
        public bool DataExists { get; set; }
        public DateTime? ServerLastStartDateTimeUTC { get; set; }
        public TimeSpan? ServerUptime { get; set; }
        public TimeSpan? LastServerUptime { get; set; }
        public DateTime? LastPulseDateTimeUTC { get; set; }
    }
}

File: src/ServerPulse.Frontend/src/app/modules/shared/domain/dtos/statistics/response/statistics/serverStatisticsResponse.ts
import { BaseStatisticsResponse, convertBaseResponse, parseDate, parseTimeSpan, TimeSpan } from "../../../../../index";

export interface ServerStatisticsResponse extends BaseStatisticsResponse {
    isAlive: boolean;
    dataExists: boolean;
    serverLastStartDateTimeUTC: Date | null;
    serverUptime: TimeSpan | null;
    lastServerUptime: TimeSpan | null;
    lastPulseDateTimeUTC: Date | null;
}

export function convertToServerStatisticsResponse(data: any): ServerStatisticsResponse {
    return {
        ...convertBaseResponse(data),
        isAlive: data?.IsAlive ?? false,
        dataExists: data?.DataExists ?? false,
        serverLastStartDateTimeUTC: parseDate(data?.ServerLastStartDateTimeUTC),
        serverUptime: parseTimeSpan(data?.ServerUptime),
        lastServerUptime: parseTimeSpan(data?.LastServerUptime),
        lastPulseDateTimeUTC: parseDate(data?.LastPulseDateTimeUTC)
    };
}